src/olympia/files/tests/test_commands.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from django.conf import settings
+from django.core.management import call_command
+
+from unittest import mock
+
+from olympia import amo
+from olympia.amo.tests import TestCase
+from olympia.addons.models import Addon
+from olympia.applications.models import AppVersion
+from olympia.files.models import File
+from olympia.files.tests.test_models import UploadMixin
+from olympia.files.utils import parse_addon
+from olympia.versions.models import Version
+from olympia.users.models import UserProfile
+
+
+class TestExtractHostPermissions(UploadMixin, TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        versions = {
+            amo.DEFAULT_WEBEXT_MIN_VERSION_NO_ID,
+            amo.DEFAULT_WEBEXT_MIN_VERSION_ANDROID,
+            amo.DEFAULT_WEBEXT_MAX_VERSION,
+        }
+        for version in versions:
+            AppVersion.objects.create(application=amo.FIREFOX.id, version=version)
+            AppVersion.objects.create(application=amo.ANDROID.id, version=version)
+
+    def setUp(self):
+        super(TestExtractHostPermissions, self).setUp()
+        self.addon = Addon.objects.create(
+            guid='mv3@extension', type=amo.ADDON_EXTENSION, name='xxx'
+        )
+        self.version = Version.objects.create(addon=self.addon)
+        UserProfile.objects.create(pk=settings.TASK_USER_ID)
+
+    def test_extract(self):
+        upload = self.get_upload('webextension_mv3.xpi')
+        parsed_data = parse_addon(upload, user=mock.Mock(groups_list=[]))
```

Comment:
`parse_addon` needs an addon passing to it if the xpi you're trying to parse has a guid defined in it's manifest.  If `addon` is not passed as an arg then the guid in the manifest is validated against all existing guids in `Addon.unfiltered` to make sure it's unique - it's failing because you're creating `self.addon` with the same guid.

Ideas: Passing `self.addon` to `parse_addon` might work?  Otherwise create `self.addon` after you've called `parse_addon`.

Reply:
Thanks!